### PR TITLE
feat(api/conf) new endpoint: UpdateStreamConnectorFile

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New configuration endpoint `UpdateStreamConnectorFile`:
- create a JSON file in the /var/lib/centreon directory with the provided content
- update the output.path configuration property with the path to the new file
- delete the previous file listed in the output.path property

Only valid for output of type stream connector (aka lua)

```
PUT /configuration/broker/<brokerId>/outputs/<outputId>/file
Body
{
 file_content: string
}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -18685,3 +18685,18 @@ msgstr "Le paramètre '%s' de type %s est invalide"
...
+msgid "Error while updating a broker output"
+msgstr "Erreur lors de la mise à jour d'une sortie Centreon Broker"
+
+msgid "Action not permitted for output of type '%s'"
+msgstr "Action non autorisée pour une sortie de type '%s'"
+
+msgid "Content is not a valid JSON string"
+msgstr "Le contenu n'est pas une chaîne de charactère JSON valide"
+
+msgid "An error occured while creating the file '%s'"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18698]:
```suggestion
msgid "An error occurred while creating the file '%s'"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -18685,3 +18685,18 @@ msgstr "Le paramètre '%s' de type %s est invalide"
...
+msgstr "Une erreur est survenue lors de la creation du fichier '%s'"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18699]:
```suggestion
msgstr "Une erreur est survenue lors de la création du fichier '%s'"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -18685,3 +18685,18 @@ msgstr "Le paramètre '%s' de type %s est invalide"
...
+
+msgid "An error occured while deleting the file '%s'"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18701]:
```suggestion
msgid "An error occurred while deleting the file '%s'"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -17340,3 +17340,18 @@ msgstr ""
...
+#  msgid "Error while updating a broker output"
+#  msgstr ""
+
+#  msgid "Action not permitted for output of type '%s'"
+#  msgstr ""
+
+#  msgid "Content is not a valid JSON string"
+#  msgstr ""
+
+#  msgid "An error occured while creating the file '%s'"
```
Comment (cg-tw) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17353]:
```suggestion
#  msgid "An error occurred while creating the file '%s'"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -17333,3 +17333,18 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+#  msgid "Error while updating a broker output"
+#  msgstr ""
+
+#  msgid "Action not permitted for output of type '%s'"
+#  msgstr ""
+
+#  msgid "Content is not a valid JSON string"
+#  msgstr ""
+
+#  msgid "An error occured while creating the file '%s'"
```
Comment (cg-tw) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17346]:
```suggestion
#  msgid "An error occurred while creating the file '%s'"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -17333,3 +17333,18 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+#  msgstr ""
+
+#  msgid "An error occured while deleting the file '%s'"
```
Comment (cg-tw) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17349]:
```suggestion
#  msgid "An error occurred while deleting the file '%s'"
```

File: centreon/src/Core/Broker/Application/Exception/BrokerException.php [hunk 7]
```diff
@@ -122,4 +130,42 @@ public static function outputNotFound(int $brokerId, int $outputId): self
...
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return self
+     */
+    public static function errorWhenCreatingFile(string $filePath): self
+    {
+        return new self(sprintf(_("An error occured while creating the file '%s'"), $filePath));
```
Comment (cg-tw) [inline centreon/src/Core/Broker/Application/Exception/BrokerException.php:159]:
```suggestion
        return new self(sprintf(_("An error occurred while creating the file '%s'"), $filePath));
```

File: centreon/src/Core/Broker/Application/Exception/BrokerException.php [hunk 8]
```diff
@@ -122,4 +130,42 @@ public static function outputNotFound(int $brokerId, int $outputId): self
...
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return self
+     */
+    public static function errorWhenDeletingFile(string $filePath): self
+    {
+        return new self(sprintf(_("An error occured while deleting the file '%s'"), $filePath));
```
Comment (cg-tw) [inline centreon/src/Core/Broker/Application/Exception/BrokerException.php:169]:
```suggestion
        return new self(sprintf(_("An error occurred while deleting the file '%s'"), $filePath));
```

File: centreon/src/Core/Broker/Infrastructure/API/UpdateStreamConnectorFile/UpdateStreamConnectorFilePresenter.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+                    $response->path,
+                    [
+                        'path' => $response->path,
+                        // 'broker_id' => $response->brokerId,
+                        // 'name' => $response->name,
+                        // 'type' => [
+                        //     'id' => $response->type->id,
+                        //     'name' => $response->type->name,
+                        // ],
+                        // 'parameters' => (object) $response->parameters,
```
Comment (a-launois) [inline centreon/src/Core/Broker/Infrastructure/API/UpdateStreamConnectorFile/UpdateStreamConnectorFilePresenter.php:56]:
```suggestion
```

File: centreon/src/Core/Broker/Infrastructure/API/UpdateStreamConnectorFile/UpdateStreamConnectorFilePresenter.php [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+                    ]
+                )
+            );
+
+            // NOT setting location as required route does not currently exist
```
Comment (a-launois) [inline centreon/src/Core/Broker/Infrastructure/API/UpdateStreamConnectorFile/UpdateStreamConnectorFilePresenter.php:61]:
```suggestion
```

**Review by cg-tw — DISMISSED** [review]

File: centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+                    new ForbiddenResponse(BrokerException::editNotAllowed()->getMessage())
+                );
+
+                return;
+            }
+
+            $this->validator->brokerIsValidOrFail($request->brokerId);
+
+            if (! ($type = $this->readOutputRepository->findType($request->type))) {
+                throw BrokerException::idDoesNotExist('type', $request->type);
```
Comment (callapa) [inline centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php:77]:
```suggestion
                throw BrokerException::TypeDoesNotExist('type', $request->type);
```

File: centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+            }
+
+            if (! ($outputFields = $this->readOutputRepository->findParametersByType($request->type, true))) {
+                throw BrokerException::idDoesNotExist('type', $request->type);
```
Comment (callapa) [inline centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php:81]:
```suggestion
                throw BrokerException::typeDoesNotExist('type', $request->type);
```

File: centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+            }
+
+            $this->validator->validateParameters($outputFields, $request->parameters);
```
Comment (callapa) [inline centreon/src/Core/Broker/Application/UseCase/AddBrokerOutput/AddBrokerOutput.php:84]: Which parameters?

**Review by cg-tw — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**